### PR TITLE
Fixes weblab report abuse link

### PR DESCRIPTION
**Pull request by hannahbergam**
The solution to this is not particularly beautiful, but it matches existing patterns and has been tested to ensure the abuse score is automatically updated for weblab projects (and reporting for other labs are unchanged). 

The base issue: cross-origin sharing is no longer allowed, but weblab projects are hosted on codeprojects and abuse reporting is hosted on studio. The way we autofill the report abuse form is through checking document.referrer, which works for all project types except weblab. 

This solution appends the channelId through a custom href created in footer.js (which was already happening in the View Code link, a solution I updated slightly away from a click event to match). 

From there, if the abuse form sees the channelId added to its url, it will use this new custom url as the abuseUrl. Otherwise, it grabs the document.referrer. I built out the getChannelIdFromUrl function to work with this new url (looking for 'channelId='), but left the other two cases so this function can still be utilized for parsing codeprojects urls if necessary.

There is a comment explaining this new logic in the code to hopefully avoid confusion.

## Links

- spec: []()
- jira ticket: []https://codedotorg.atlassian.net/jira/software/c/projects/LP/boards/21?modal=detail&selectedIssue=LP-2446

## Testing story

I tested this locally, making sure the url is piped through with the channel id, and making sure that actually reporting it will increment the abuse score on a weblab project.

I also tested that report abuse still works correctly with other project types and tested that the abuse score is updated by reporting a local gamelab project.

I also tested the new regex logic in ReportAbuseFormTest to make sure the channelId is retrieved properly from the new custom urls created.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by epeach — APPROVED** [review]
This looks like a great way to follow an established pattern to achieve our goals!

File: apps/src/reportAbuse.js [hunk 1]
```diff
@@ -19,6 +19,8 @@ export const getChannelIdFromUrl = function(abuseUrl) {
   let match;
   if (abuseUrl.indexOf('codeprojects') >= 0 && abuseUrl.indexOf('weblab') < 0) {
     match = /.*codeprojects.*[^\/]+\/([^\/]+)/.exec(abuseUrl);
+  } else if (abuseUrl.indexOf('channelId=') >= 0) {
+    match = /\?channelId=(.*)/.exec(abuseUrl);
```
Comment (bethanyaconnor) [inline apps/src/reportAbuse.js:23]: Can you add a test in apps/test/unit/code-studio/components/ReportAbuseFormTest.js that tests this new logic?

File: dashboard/public/weblab/footer.js [hunk 2]
```diff
@@ -29,11 +29,11 @@ var applyFooter = function() {
     document.body.appendChild(footerContents.childNodes[0]);
   }
   if (window.encrypted_channel_id) {
-    var link = document.getElementById("pagefooter_view_code");
-    link.onclick = function() {
-      window.location.href =
-        link.href + "/" + window.encrypted_channel_id + "/view";
-      return false;
-    };
+    var viewCodeLink = document.getElementById("pagefooter_view_code");
```
Comment (bethanyaconnor) [inline dashboard/public/weblab/footer.js:32]: This is still so much 🍝 but still better than before 😂 Thanks for working through this gnarly part of our system!

**Review by bethanyaconnor — APPROVED** [review]
One ask for a test but otherwise looks good!

File: apps/src/reportAbuse.js (hunk 1 above)
Comment (hannahbergam) [reply apps/src/reportAbuse.js:23]: Ahh yes I should've done this initially-thank you. Done!

File: dashboard/public/weblab/footer.js (hunk 2 above)
Comment (hannahbergam) [reply dashboard/public/weblab/footer.js:32]: 😂🤦🏻‍♀️ so true. And thanks for the help!

**Review by bethanyaconnor — APPROVED** [review]

**Pull request merged**